core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -857,22 +870,31 @@ class TiBatchWrite(@transient val df: DataFrame,
 
         // region split
         if (regionSplitNum > 1) {
-          val minHandle = rdd.min().handle
-          val maxHandle = rdd.max().handle
-          if (checkTidbRegionSplitContidion(minHandle, maxHandle, regionSplitNum) || options.regionSplitNum != 0) {
-            logger.info("region split num=" + regionSplitNum + " index name=" + index.getName)
+          val minIndexValue = rdd.min()(ordering).row.get(colOffset, dataType).toString
+          val maxIndexValue = rdd.max()(ordering).row.get(colOffset, dataType).toString
+          logger.info(
+            s"index region split, regionSplitNum=$regionSplitNum, indexName=${index.getName}"
+          )
+          try {
             tiDBJDBCClient
               .splitIndexRegion(
                 options.database,
                 options.table,
                 index.getName,
-                minHandle,
-                maxHandle,
+                minIndexValue,
+                maxIndexValue,
                 regionSplitNum
               )
-          } else {
-            logger.warn("region split is skipped")
+          } catch {
+            case e: SQLException =>
+              if (options.isTest) {
```

Comment:
if this is a debug-only parameter, should it be caught and thrown somewhere else? e.g., in the test framework